go/src/client/main.go

```diff
@@ -328,15 +330,15 @@ func nextGame(httpClient *http.Client) error {
 			return err
 		}
 		result, pgn := playMatch(networkPath, candidatePath, params, nextGame.Flip)
-		client.UploadMatchResult(httpClient, *HOSTNAME, nextGame.MatchGameId, result, pgn, getExtraParams())
+		go client.UploadMatchResult(httpClient, *HOSTNAME, nextGame.MatchGameId, result, pgn, getExtraParams())
 		return nil
 	} else if nextGame.Type == "train" {
 		networkPath, err := getNetwork(httpClient, nextGame.Sha, true)
 		if err != nil {
 			return err
 		}
 		trainFile, pgn := train(networkPath, params)
```

Comment:
We could potentially even read in the trainFile into memory, and return it from `train`.  That way we don't have to worry about the race here.

Reply:
I don't understand what the race is, since the path is different each time this is called

Reply:
The path is the same each time isn't it though?  `train-{pid}/training.0.gz` type of thing.  Process id remains the same (it's the process ID of the client process).

Reply:
@Error323 Can we use the random seed that was used an name each file training.+seed+.gz or will that cause problems with your training? Should fix the problem of duplicate paths and have the bonus that in a crash, we could upload result without losing it in memory.

Reply:
Actually training.0.gz is hardcoded in main.go with no access to seed, but I could use a new rnd call with little work.